nextest-runner/src/runner/dispatcher.rs

```diff
@@ -106,6 +112,9 @@ where
 
         loop {
             let internal_event = tokio::select! {
+                _ = tokio::time::sleep_until(self.timeout_at.into()) => {
```

Comment:
This should use a [`PausableSleep`](https://github.com/nextest-rs/nextest/blob/7483c7730e2b62e0b50488e3dc3962cb9dbdd9d5/nextest-runner/src/time/pausable_sleep.rs) so that the timeout isn't taking effect while the test run is paused.

Reply:
When should it be paused/resumed? In `handle_signal_event`'s `SignalEvent::JobControl(JobControlEvent::Stop)` and `Continue` ?

Reply:
Actually realized I can do it as a reaction to `HandleEventResponse::JobControl` messages in the same function that does the sleep.